### PR TITLE
Benchmark performance of MIB on various storage media

### PR DESCRIPTION
**Pull request by uellue**
Refs #838

Preliminary results:

<details>

```

----------------------------------------------------------------------------------------------------------------------------- benchmark: 18 tests ------------------------------------------------------------------------------------------------------------------------------
Name (time in s)                                                                                                      Min                Max               Mean            StdDev             Median               IQR            Outliers     OPS            Rounds  Iterations
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_mask[/cachedata/users/libertem/reference/-cold_cache-cold_executor]                                           9.6043 (57.13)     9.6043 (57.13)     9.6043 (57.13)    0.0000 (1.0)       9.6043 (57.13)    0.0000 (1.0)           0;0  0.1041 (0.02)          1           1
test_mask[/cachedata/users/libertem/reference/-cold_cache-warm_executor]                                           0.6600 (3.93)      0.6600 (3.93)      0.6600 (3.93)     0.0000 (1.0)       0.6600 (3.93)     0.0000 (1.0)           0;0  1.5152 (0.25)          1           1
test_mask[/cachedata/users/libertem/reference/-warm_cache-cold_executor]                                           9.5442 (56.77)     9.5442 (56.77)     9.5442 (56.77)    0.0000 (1.0)       9.5442 (56.77)    0.0000 (1.0)           0;0  0.1048 (0.02)          1           1
test_mask[/cachedata/users/libertem/reference/-warm_cache-warm_executor]                                           0.5726 (3.41)      0.5726 (3.41)      0.5726 (3.41)     0.0000 (1.0)       0.5726 (3.41)     0.0000 (1.0)           0;0  1.7465 (0.29)          1           1
test_mask[/data/users/libertem/reference/-cold_cache-cold_executor]                                               11.1324 (66.22)    11.1324 (66.22)    11.1324 (66.22)    0.0000 (1.0)      11.1324 (66.22)    0.0000 (1.0)           0;0  0.0898 (0.02)          1           1
test_mask[/data/users/libertem/reference/-cold_cache-warm_executor]                                                3.3322 (19.82)     3.3322 (19.82)     3.3322 (19.82)    0.0000 (1.0)       3.3322 (19.82)    0.0000 (1.0)           0;0  0.3001 (0.05)          1           1
test_mask[/data/users/libertem/reference/-warm_cache-cold_executor]                                                9.9714 (59.31)     9.9714 (59.31)     9.9714 (59.31)    0.0000 (1.0)       9.9714 (59.31)    0.0000 (1.0)           0;0  0.1003 (0.02)          1           1
test_mask[/data/users/libertem/reference/-warm_cache-warm_executor]                                                0.4795 (2.85)      0.4795 (2.85)      0.4795 (2.85)     0.0000 (1.0)       0.4795 (2.85)     0.0000 (1.0)           0;0  2.0854 (0.35)          1           1
test_mask[/storage/holo/clausen/testdata/ER-C-1/groups/data_science/data/reference/-cold_cache-cold_executor]      9.8506 (58.59)     9.8506 (58.59)     9.8506 (58.59)    0.0000 (1.0)       9.8506 (58.59)    0.0000 (1.0)           0;0  0.1015 (0.02)          1           1
test_mask[/storage/holo/clausen/testdata/ER-C-1/groups/data_science/data/reference/-cold_cache-warm_executor]      1.2952 (7.70)      1.2952 (7.70)      1.2952 (7.70)     0.0000 (1.0)       1.2952 (7.70)     0.0000 (1.0)           0;0  0.7721 (0.13)          1           1
test_mask[/storage/holo/clausen/testdata/ER-C-1/groups/data_science/data/reference/-warm_cache-cold_executor]      9.4739 (56.35)     9.4739 (56.35)     9.4739 (56.35)    0.0000 (1.0)       9.4739 (56.35)    0.0000 (1.0)           0;0  0.1056 (0.02)          1           1
test_mask[/storage/holo/clausen/testdata/ER-C-1/groups/data_science/data/reference/-warm_cache-warm_executor]      0.4948 (2.94)      0.4948 (2.94)      0.4948 (2.94)     0.0000 (1.0)       0.4948 (2.94)     0.0000 (1.0)           0;0  2.0209 (0.34)          1           1
test_sequential[/cachedata/users/libertem/reference/-cold_cache]                                                   0.3637 (2.16)      0.3637 (2.16)      0.3637 (2.16)     0.0000 (1.0)       0.3637 (2.16)     0.0000 (1.0)           0;0  2.7499 (0.46)          1           1
test_sequential[/cachedata/users/libertem/reference/-warm_cache]                                                   0.1681 (1.0)       0.1681 (1.0)       0.1681 (1.0)      0.0000 (1.0)       0.1681 (1.0)      0.0000 (1.0)           0;0  5.9484 (1.0)           1           1
test_sequential[/data/users/libertem/reference/-cold_cache]                                                        2.7815 (16.55)     2.7815 (16.55)     2.7815 (16.55)    0.0000 (1.0)       2.7815 (16.55)    0.0000 (1.0)           0;0  0.3595 (0.06)          1           1
test_sequential[/data/users/libertem/reference/-warm_cache]                                                        0.1802 (1.07)      0.1802 (1.07)      0.1802 (1.07)     0.0000 (1.0)       0.1802 (1.07)     0.0000 (1.0)           0;0  5.5503 (0.93)          1           1
test_sequential[/storage/holo/clausen/testdata/ER-C-1/groups/data_science/data/reference/-cold_cache]              1.1418 (6.79)      1.1418 (6.79)      1.1418 (6.79)     0.0000 (1.0)       1.1418 (6.79)     0.0000 (1.0)           0;0  0.8758 (0.15)          1           1
test_sequential[/storage/holo/clausen/testdata/ER-C-1/groups/data_science/data/reference/-warm_cache]              0.1880 (1.12)      0.1880 (1.12)      0.1880 (1.12)     0.0000 (1.0)       0.1880 (1.12)     0.0000 (1.0)           0;0  5.3179 (0.89)          1           1
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

```

</details>

* Linux doesn't seem to be affected by #838, possibly because of using posix_fadvise()?
* Warming up and dropping caches like this works on Linux

TODOs:
- [x] Drop caches on Windows to reproduce there
- [x] Better solution for storage location per machine -- environment variables, configuration file, ...?

## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added/updated test cases

### CODE REVIEW

**Comment by uellue** [comment]
@sk1p the CI failure with error 500 on Py36 happened already before. It seems unrelated to this change. Any idea?

**Comment by sk1p** [comment]
> the CI failure with error 500 on Py36 happened already before. It seems unrelated to this change. Any idea?

Sadly I haven't managed to find the cause for this one - the `distributed` code around where that exception is thrown notes "track down infrequent error". Conceptually, this could happen if, for some reason, the event loop fails to start within the given timeout (10s here). So this could possibly be caused by the CI system being generally overloaded, but without any more data I'm not sure.

**Comment by uellue** [comment]
Through `win32file` in `pywin32` and `msvcrt` one can use the Windows API to do "funny things" with files, including possibly Direct IO.

Here an example on how to create a Windows file handle with custom flags, and then turn this into a Python file object:

```python
import win32file
import msvcrt
import os

path = r'C:\Users\weber\Largedata\LargeData\ER-C-1\projects\ptycho-4.0\data\Readme.md'

f = win32file.CreateFile(
    path,
    win32file.GENERIC_READ,
    win32file.FILE_SHARE_READ,
    None,
    win32file.OPEN_EXISTING,
    win32file.FILE_FLAG_NO_BUFFERING,  # this is Direct IO
    0,
)

pyfile = os.fdopen(msvcrt.open_osfhandle(f.handle, os.O_RDONLY))

# This may not be the right block size for direct IO, further testing required...
pyfile.read(1000)
```

Microsoft has more info on that, if required. :-)

File: src/libertem/utils/threading.py [hunk 1]
```diff
@@ -26,6 +31,23 @@ def set_fftw_threads(n):
     def set_fftw_threads(n):
         yield
 
+if torch:
+    @contextmanager
+    def set_torch_threads(n):
+        torch_threads = torch.get_num_threads()
+        # torch_interop_threads = torch.get_num_interop_threads()
```
Comment (sk1p) [inline src/libertem/utils/threading.py:38]: Could you add a note why the `*_interop_*` calls are commented out?

**Review by sk1p — COMMENTED** [review]
Thanks, benchmarks look good, added a comment about the torch thread control thingy. Could you maybe add a link to the relevant documentation for the `drop_cache` win32 function?

Thanks!

**Review by sk1p — APPROVED** [review]

**Pull request merged**